### PR TITLE
Add diff based execution

### PR DESCRIPTION
**Pull request by EmanueleMinotto**
This pull request adds the option `--must-match-git-diff` to the process command, this is useful for both local and CI execution.

It should be compatible with the rest of the configuration because creates an intersection with the `git diff` output.

### CODE REVIEW

**Comment by TomasVotruba** [comment]
Rebase on `master`

**Comment by EmanueleMinotto** [comment]
@TomasVotruba done

**Comment by TomasVotruba** [comment]
Thank, can you update `README` with this feature in 2-3 lines?

Right above: https://github.com/rectorphp/rector#3-steps-to-create-your-own-rector

File: README.md [hunk 1]
```diff
@@ -201,6 +201,13 @@ parameters:
     auto_import_names: false
 ```
 
+### Limit Execution to Changed Files
+
+Execution can be limited to changed files using the `process` option `--must-match-git-diff`.
+This option will filter the files included by the configuration, creating an intersection with the files listed in `git diff`.
+
```
[the remark this answers is not in the document]
Comment (TomasVotruba) [reply README.md]: I'll handle that locally, let's merge it

**Pull request merged**